test/teststl.cpp

```diff
@@ -5200,6 +5200,16 @@ class TestStl : public TestFixture {
               "}\n",
               true);
         ASSERT_EQUALS("", errout.str());
+
+        check("bool g(int);\n"
+              "int f(const std::vector<int>& v) {\n"
+              "    int ret = 0;\n"
+              "    for (const auto i : v)\n"
+              "        if (!g(i))\n"
+              "            ret = 1;\n"
```

Comment:
will this still warn when doing `ret = ret + 1` or `ret = h(ret)`? We should probably check that the variable appears on the RHS.

Reply:
We suggest `std::count_if` for `ret = ret + 1`. There is no warning for `ret = h(ret)` with or without this change.